pkg/util/proto/document_v3.go

```diff
@@ -0,0 +1,329 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proto
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	openapi_v3 "github.com/google/gnostic/openapiv3"
+	"gopkg.in/yaml.v3"
+)
+
+// Temporary parse implementation to be used until gnostic->kube-openapi conversion
+// is possible.
+func NewOpenAPIV3Data(doc *openapi_v3.Document) (Models, error) {
+	definitions := Definitions{
+		models: map[string]Schema{},
+	}
+
+	components := doc.GetComponents()
+	if components == nil {
+		return &definitions, nil
+	}
+
+	schemas := components.GetSchemas()
+	if schemas == nil {
+		return &definitions, nil
+	}
```

Comment:
Maybe I'm wrong, but if gnostic still works the same, it usually works well when you call methods on nil pointers. So you can shorten that to:
```suggestion
	schemas := components.GetComponents().GetSchemas()
	if schemas == nil {
		return &definitions, nil
	}
```